source/blog/2018-06-21-ember-3-2-released.md

```diff
@@ -0,0 +1,350 @@
+
+But, when run normally, calls to its methods must not be bound to anything other than  the console object. If not, you will receive an Invalid calling object exception. This is a known inconsistency with these browsers.
+
+To avoid this, transform this:
+
+```javascript
+var print = Logger.log; // assigning method to variable
+```
+
+into this:
+
+```javascript
+// assigning method bound to console to variable
+var print = console.log.bind(console);
+```
+
+Also, transform any of the following:
+
+```javascript
+Logger.info.apply(undefined, arguments); // or
+Logger.info.apply(null, arguments); // or
+Logger.info.apply(this, arguments); // or
+```
+
+into this:
+
+```javascript
+console.info.apply(console, arguments);
+```
+
+Finally, because node versions before version 9 don't support console.debug, you may want to transform the following:
+
+```javascript
+Logger.debug(message);
+```
+
+into this:
+
+```javascript
+if (console.debug) {
+  console.debug(message);
+} else {
+  console.log(message);
+}
+```
+
+**Note for Add-on Authors** - If your add-on needs to support both Ember 2.x and Ember 3.x clients, you will need to test for the existence of console before calling its methods. If you do much logging, you may find it convenient to define your own wrapper. Writing the wrapper as a service will provide for dependency injection by tests and perhaps even clients.
+
+##### Private API Router#route renamed (2 of 3)
+
+In order to avoid collisions with user-defined properties or methods, the `Router#route` private API has been renamed to `Router#_route`. If you want access to the router, you should inject the router service into the route like this:
+
+```javascript
+import Route from '@ember/routing/route';
+import { inject as service } from '@ember/service';
+
+export default Route.extend({
+  router: service()
+});
+```
+
+##### Define computed properties with defineProperty (3 of 3)
+
+Use `defineProperty` to define computed properties. Although uncommon, it is possible to assign computed properties directly to objects. This way they are implicitly computed from things like `Ember.get`. Assigning computed properties directly is deprecated to support ES5 getter computed properties, and you should replace these assignments with calls to `defineProperty`.
+
+For example, the following:
+
+```javascript
+let object = {};
+object.key = Ember.computed(() => 'value');
+Ember.get(object, 'key') === 'value';
+```
+
+Should be changed to this:
+
+```javascript
+let object = {};
+Ember.defineProperty(object, 'key', Ember.computed(() => 'value'));
+Ember.get(object, 'key') === 'value';
+```
+
+For more details on changes in Ember.js 3.2, please review the [Ember.js 3.2.0 release page](https://github.com/emberjs/ember.js/releases/tag/v3.2.0).
```

Comment:
The blog post should also mention that due to some internal changes in Ember.js the `@ember/test-helpers` dependency needs to be updated to function properly. Otherwise people will experience tests not waiting until requests are resolved and similar issues.

Reply:
What specific change is required?

Reply:
to the blog post or to the projects?

if the projects use a lockfile they will have to somehow update the subdependency. I usually do that by hand-editing it out of the lockfile and run `yarn` again, but there might be better ways.

Reply:
Sorry, I mean, which change in Ember requires a bump of @ember/test-helpers (I’m just forgetting)?

Reply:
the require path of `checkWaiters()` had changed

Reply:
Thank you!!